From c1eff086c5392337ac45af925776f2e4418da22a Mon Sep 17 00:00:00 2001
From: Nagaraj S <sivasankaranx.nagaraj@intel.com>
Date: Fri, 1 Apr 2016 14:57:47 -0700
Subject: [PATCH 378/441] mailbox:throw warning if ATOM delay to receive RPC

wait_for_completion() function put the process to D state
i.e uninterruptable sleep. If RPC message is not arrived within 120 sec
then ce_mailbox driver throw the warning
(blocked the process more than 120 sec).

The solution is to put the process to S state (i.e interruptable sleep).
So change the API to wait_for_completion_interruptable().

Change-Id: Ia281135daa4a56cb88e8b9c9bcdd9e7a249aca30
---
 drivers/misc/ce_mailbox.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/drivers/misc/ce_mailbox.c
+++ b/drivers/misc/ce_mailbox.c
@@ -165,8 +165,10 @@ static int mbx_close(struct inode *inode
 /* Polloing on specivec callback from mailbox - max wait forever to get RPC info from ARM  */
 static int mbx_wait_till_ready()
 {
-	wait_for_completion(&cmd_done);
-        return 0;
+	int res = wait_for_completion_interruptible(&cmd_done);
+	if (res)
+	    printk("CE_MAILBOX: %s:%u: interrupted \n", __func__, __LINE__ );
+	return res;
 }
 
  /* * npcpu_appcpu_mbx_receive_callback
